pkg/template/nstemplatetiers/nstemplatetier_generator_test.go

```diff
@@ -280,19 +277,29 @@ func TestGenerateTiers(t *testing.T) {
 			// when
 			err := GenerateTiers(s, ensureObjectFuncForClient(clt), namespace, getTestMetadata(), getTestTemplates(t))
 			require.NoError(t, err)
+			// here we're just capturing the state after the first call so that we can compare with the state after the second call
+			tierTmpls := toolchainv1alpha1.TierTemplateList{}
+			err = clt.List(context.TODO(), &tierTmpls, runtimeclient.InNamespace(namespace))
+			require.NoError(t, err)
+			require.Len(t, tierTmpls.Items, 16) // 4 items for advanced and base tiers + 3 for nocluster tier + 4 for appstudio
+			origTemplates := map[string]*toolchainv1alpha1.TierTemplate{}
+			for _, tmpl := range tierTmpls.Items {
+				origTemplates[tmpl.Name] = &tmpl
+			}
 
 			// when calling CreateOrUpdateResources a second time
 			err = GenerateTiers(s, ensureObjectFuncForClient(clt), namespace, getTestMetadata(), getTestTemplates(t))
 
 			// then
 			require.NoError(t, err)
 			// verify that all TierTemplate CRs were updated
-			tierTmpls := toolchainv1alpha1.TierTemplateList{}
+			tierTmpls = toolchainv1alpha1.TierTemplateList{}
 			err = clt.List(context.TODO(), &tierTmpls, runtimeclient.InNamespace(namespace))
 			require.NoError(t, err)
 			require.Len(t, tierTmpls.Items, 16) // 4 items for advanced and base tiers + 3 for nocluster tier + 4 for appstudio
+			// check that the tier templates are unchanged
 			for _, tierTmpl := range tierTmpls.Items {
-				assert.Equal(t, int64(1), tierTmpl.Generation) // unchanged
+				assert.True(t, reflect.DeepEqual(origTemplates[tierTmpl.Name].Spec, tierTmpl.Spec))
```

Comment:
You could do 
```suggestion
				assert.Equal(t, origTemplates[tierTmpl.Name].Spec, tierTmpl.Spec)
```
instead, right?
Or why would the generation change here? The tier is not going to be updated for the same template, right? Basically why you need to change the test here at all?

Reply:
Nice investigation!